extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -75,12 +114,92 @@ export class AmpConsent extends AMP.BaseElement {
           }
         });
 
-    getServicePromiseForDoc(this.getAmpDoc(), CONSENT_STATE_MANAGER)
-        .then(manager => {
-          this.consentStateManager_ = manager;
+    const consentStateManagerPromise =
+        getServicePromiseForDoc(this.getAmpDoc(), CONSENT_STATE_MANAGER)
+            .then(manager => {
+              this.consentStateManager_ = manager;
+            });
+    const notificationUiManagerPromise =
+        getServicePromiseForDoc(this.getAmpDoc(), NOTIFICATION_UI_MANAGER)
+            .then(manager => {
+              this.notificationUiManager_ = manager;
+            });
+
+    Promise.all([consentStateManagerPromise, notificationUiManagerPromise])
+        .then(() => {
           this.init_();
         });
+  }
+
+  /**
+   * To show prompt UI for instanceId
+   * @param {string} instanceId
+   * @return {!Promise}
+   */
+  show_(instanceId) {
+    if (!this.consentUIRequired_[instanceId] || !this.consentUI_[instanceId]) {
+      return Promise.resolve();
+    }
+
+    dev().assert(!this.currentDisplayInstance_,
+        'Other consent instance on display');
+
+
+    if (!this.uiInit_) {
+      this.uiInit_ = true;
+      toggle(this.element, true);
+      this.getViewport().addToFixedLayer(this.element);
+    }
+
+    this.element.classList.remove('amp-hidden');
+    this.element.classList.add('amp-active');
+
+    // Display the current instance
+    this.currentDisplayInstance_ = instanceId;
+    setStyle(this.consentUI_[this.currentDisplayInstance_], 'display', 'block');
+    return new Promise(resolve => {
+      this.dialogResolver_[instanceId] = resolve;
```

Comment:
I wonder if all these promises/resolvers are necessary. Would it be simpler/possible to have a synchronous queue that just gets updated on button click?

Reply:
Talked offline